src/toplevel.c

```diff
@@ -362,7 +362,7 @@ static void expr_attributes(jl_value_t *v, int *has_intrinsics, int *has_defs)
             jl_module_t *mod = jl_globalref_mod(f);
             jl_sym_t *name = jl_globalref_name(f);
             if (jl_binding_resolved_p(mod, name)) {
-                jl_binding_t *b = jl_get_binding(mod, name);
+                jl_binding_t *b = jl_get_module_binding(mod, name);
```

Comment:
This caused part of #31965. What was the reason for this?

Reply:
Don't quite remember, but `jl_get_module_binding` is NOTSAFEPOINT, so I suspect something was getting accidentally gc'ed on the get_binding call, and I may have misunderstood the difference between these functions